### ISSUE TITLE
Notification not Fires in iOS

### ISSUE THREAD

**Issue by pooya511** [phonegap/phonegap-plugin-push#631]
Hi,

I followed the instruction step by step. Plugin is attached to project successfully and when I open the app it asks for permission. Fortunately I can get registration ID. For server side sender I use [APN Tester free](https://itunes.apple.com/us/app/apn-tester-free/id626590577?mt=12). I import certificate (generated in App-ID section of my developer account under development part of Push Notification access) and test it with this paload:
{"aps":{"alert":"Hello World!","badge":1}}

I got confused. I don't know where the problem is. I double checked certificate and provisioning profile.

Thanks

**Comment by matdev** [comment]
Do you send the notification to the device token you get when your `push.on('registration',...)` event is fired i.e. data.registrationId ?

Do you see the notification in iOS notification center ?

**Comment by pooya511** [comment]
Currently i copy paste device reg ID which is 32 characters into apns online tester. I'm want to use APNs directly and not to use CGM because it has limited functionality.


Pooya Sarraf

**Comment by pooya511** [comment]
I use following code in my application. I also use development certificate,development provisioning profile and development APNs ssl certificate(I think this is sandbox). 
I guess the application can't diagnose to use **sandbox ** server and us main server. How can I tell the app to use sandbox?


var push_id;
    var push = PushNotification.init({
        "ios": {
            "senderID": "475058352589",
            gcmSandbox: true,
            alert: "true",
            badge: true,
            sound: 'true'
        }
    });

    push.on('registration', function (data) {
        // send data.registrationId to push service
        push_id = data.registrationId;
        alert(data.registrationId);
    });


    push.on('notification', function (data) {
        // do something with the push data
        // then call finish to let the OS know we are done
        alert(data);
        push.finish(function () {
            console.log("processing of push data is finished");
        });
    });

**Comment by pooya511** [comment]
Problem solved. I noticed Bundle Identifier should be as same as provisioning profile like com.something.something2 . That was not on any of documentations and I spent over a week :)

**Comment by matdev** [comment]
Glad you solved your problem.

For info, you don't need to set the gcmSandbox flag in the ios init section.
You tell the app to use sandbox when you set a developement provisioning profile in XCode

**Comment by macdonst** [comment]
@matdev no, you do have to set the gcmSandbox property. There is code in the PushPlugin that needs that parameter set properly.

**Comment by barocsi** [comment]
I have set all certs properly. 
GCMSandbox property is set. 
Provisioning is development. 

When testing manually APNs its working. Only GCM that is not able to peer through. Bundle ID is set properly.

I wonder if this is switched by gcmsandbox:true :
https://developers.google.com/cloud-messaging/ios/certs
```
In your code, make sure that the value of the kGGLInstanceIDAPNSServerTypeSandboxOption option that you are passing to GGLInstanceID.tokenWithAuthorizedEntity:scope:options:handler: is correctly set for the environment that you are using: true for development, false for production. If you are not using the correct environment, messages will not be delivered to your app.```

In cordova I can see that the phone is registering in sandbox mode, I can send message using this:
```
curl --header "Authorization: key=xxx" --header "Content-Type: application/json" https://android.googleapis.com/gcm/send -d"{\"registration_ids\":[\"yyy\"], \"collapse_key\": \"Turn\", \"content_available\": true, \"notification\": { \"title\": \"title\", \"body\": \"body\", \"badge\": \"1\" }}```

The response is success.

However the device is not receiving anything.

**Comment by bluejamesbond** [comment]
@barocsi same situation here
